Yoast/javascript
# Improve the way keyphrase distribution assessment works with lists

## Summary


This PR can be summarized in the following changelog entry:

* [yoastseo] Makes sure that lists with single words don't skew the keyphrase distribution score.
* [Yoast SEO Premium] Fixes a bug when the keyphrase distribution assessment might return skewed results if a text contains lists of single words.

## Relevant technical choices:
* I've chosen a very straightforward solution, simply removing all list tags from the text before it gets processed further for keyphrase distribution.
* One drawback of the new functionality is that list items that are sentences and don't end in a punctuation mark now are merged with the following list item. In longer lists, this can lead to very long sentences, which means that false positives might be likely. However, this will lead to a keyphrase distribution result that is too positive in some cases. While not a completely accurate result, it will not cause as frustration for the user as an undeserved negative result.

## Test instructions

This PR can be tested by following these steps:

* Have a look at the specs. Think about whether you can come up with any scenarios not yet covered.
* Add a relatively short text to your paper (but long enough to trigger keyphrase distribution). Add a keyphrase and instances of that keyphrase in the text. Add a list with single words as list items. Make sure that your keyphrase distribution score doesn't get worse after adding the list of single words.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  *

## UI changes

## Quality assurance


Fixes https://yoast.atlassian.net/browse/LIN-333
